moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp

```diff
@@ -286,54 +301,62 @@ bool TrajectoryExecutionManager::push(const moveit_msgs::RobotTrajectory& trajec
         ss << trajectory_part << std::endl;
       ROS_INFO_NAMED(LOGNAME, "%s", ss.str().c_str());
     }
-    trajectories_.push_back(context);
+    trajectories_.push_back(std::move(context));
     return true;
   }
   else
   {
-    delete context;
+    context.reset();
     last_execution_status_ = moveit_controller_manager::ExecutionStatus::ABORTED;
   }
 
   return false;
 }
 
-bool TrajectoryExecutionManager::pushAndExecute(const moveit_msgs::RobotTrajectory& trajectory,
-                                                const std::string& controller)
+bool TrajectoryExecutionManager::pushAndExecuteSimultaneously(const moveit_msgs::RobotTrajectory& trajectory,
```

Comment:
I think nobody is using `pushAndExecute`, as far as I know, the current version is broken. So, maybe it would be a good opportunity to clean up some or most of the overload methods that may not be used in the new API.